src/MetadataViewer/MetadataViewerProvider.ts

```diff
@@ -0,0 +1,225 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+/*
+Some part of this code refers to
+https://github.com/microsoft/vscode-extension-samples/blob/2556c82cb333cf65d372bd01ac30c35ea1898a0e/custom-editor-sample/src/catScratchEditor.ts
+*/
+
+import * as vscode from 'vscode';
+import { Node } from '../OneExplorer/OneExplorer';
+import { MetadataViewer } from './MetadataViewer';
+
+
+export class RelationViewerDocument implements vscode.CustomDocument {
+  private readonly _uri: vscode.Uri;
+  private _metadataViwer: MetadataViewer[];
+  
+
+  static async create(uri: vscode.Uri):
+      Promise<RelationViewerDocument|PromiseLike<RelationViewerDocument>> {
+    return new RelationViewerDocument(uri);
+  }
+
+  private constructor(uri: vscode.Uri) {
+    this._uri = uri;
+    this._metadataViwer = [];
+    
+  }
+
+  public get uri() {
+    return this._uri;
+  }
+
+  // CustomDocument implements
+  dispose(): void {
+    // NOTE panel is closed before document and this is just for safety
+    this._metadataViwer.forEach((view) => {
+      while (this._metadataViwer.length) {
+        view.disposeMetadataView();
+      }
+    });
+    this._metadataViwer = [];
+  }
+
+  public openView(panel: vscode.WebviewPanel, extensionUri: vscode.Uri, fileUri:vscode.Uri) {
+    let view = new MetadataViewer(panel,extensionUri);
+
+    view.initMetadataInfo();
+    view.loadContent();
+    this._metadataViwer.push(view);
+
+    //메타데이터 정보를 가져오는 로직(Uri 인자를 이용하면 됨)
```

Comment:
Could you write the comments in English? We recommend all committers to do so, because some of us use different languages 😄

Reply:
Oh, I'm sorry. I`ll try to change it to English at once just before the end of the project, but I`m going to write it down in English right away.